### PR TITLE
feat: add token update

### PR DESCRIPTION
**Pull request by lcschy**
## Description

- Adds `tokens.update` method for tokens client.

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

- [x] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: src/types/models/tokens.ts [hunk 1]
```diff
@@ -59,9 +59,17 @@ type CreateToken<DataType = Primitive> = Pick<
   deduplicateToken?: boolean;
 };
 
+type UpdateToken<DataType = Primitive> = Partial<
+  Pick<
+    Token<DataType>,
+    'data' | 'metadata' | 'searchIndexes' | 'fingerprintExpression'
+  > & { privacy: Omit<TokenPrivacy, 'classification'> }
```
Comment (koalaty-code) [inline src/types/models/tokens.ts:66]: Do we need to add `deduplicateToken` to the update model as a nullable boolean?

File: src/types/models/tokens.ts [hunk 2]
```diff
@@ -59,9 +59,20 @@ type CreateToken<DataType = Primitive> = Pick<
   deduplicateToken?: boolean;
 };
 
+type UpdateToken<DataType = Primitive> = Partial<
+  Pick<
+    Token<DataType>,
+    'data' | 'metadata' | 'searchIndexes' | 'fingerprintExpression'
+  > & {
+    privacy: Omit<TokenPrivacy, 'classification'>;
```
Comment (djejaquino) [inline src/types/models/tokens.ts:67]:
```suggestion
    privacy?: Omit<TokenPrivacy, 'classification'>;
```
Wouldn't this be optional?

Comment (lcschy) [reply src/types/models/tokens.ts:67]: Everything is optional AFAIK, have the object wrapped on a `Partial`

Comment (lcschy) [reply src/types/models/tokens.ts:67]: I took off the other ? from that object to avoid confusion

Comment (djejaquino) [reply src/types/models/tokens.ts:71]: oh yeah. `Partial` should do the job

**Review by djejaquino — DISMISSED** [review]

File: src/tokens/BasisTheoryTokens.ts [hunk 3]
```diff
@@ -43,6 +43,24 @@ export const BasisTheoryTokens = new CrudBuilder(
...
+    public update(
+      id: string,
+      model: UpdateToken,
+      options: RequestOptions = {}
+    ): Promise<Token> {
+      const url = `/${id}`;
+
+      return this.client
+        .patch(url, model, {
+          ...createRequestConfig(options),
```
Comment (djejaquino) [inline src/tokens/BasisTheoryTokens.ts:55]: Call `createRequestConfig` once and store in a variable

**Review by djejaquino — APPROVED** [review]

**Pull request merged**